tests/sessions/retry.txtar

```diff
@@ -0,0 +1,47 @@
+None
+0 test1
+1 test1
+2 test1
+[None, "test1"]
+0 test2
+1 test2
+2 test2
+[None, "test2"]
+2s
+0 test3 [0] 0
+1 test3 [0, 1] 2
+2 test3 [0, 1, 2] 6
+3 test3 [0, 1, 2, 3] None
+[None, "test3"]
+10s
+
+-- main.star:main --
+def immediate_policy(err, attempt):
+    print(attempt, err)
+    return attempt < 2
+
+def delayed_policy(err, attempt):
+    print(attempt, err)
+    return "1s" if attempt < 2 else None
```

Comment:
I prefer that instead of a string you'll return a duration (or timedelta as it's called in Python).